docs-tech/bakery-architecture.md

```diff
@@ -52,3 +52,74 @@ and are pure computations, which traditionally are called "sieves".
 A library to be included within the consumers of bakery, to ease the integration of old and new baker users. 
 It exposes the traditional baker java and scala api, but within its implementation it requests through http 
 to the exposed baker clusters. One instance of the client is created per baker cluster.
+
+
+# Bakery Kubernetes Implementation
+
+![Kubernetes diagram](./Bakery%20Kubernetes%20Architecture.png)
+
+## Bakery Controller
+
+The Bakery Controller follows Kubernete's general architecture pattern called [Controller Pattern](https://kubernetes.io/docs/concepts/architecture/controller/)
+It is deployed as a single pod which uses Kubernete's API to watch the Baker and Interaction CRDs, which have the
+specification of the Baker clusters and Interactions to be deployed. The Baker Controller constantly watches for creation,
+changes and deletions of such CRDs, so that the state of the namespace always matches the textual specifications.
+
+The Bakery Controller also creates Kubernetes ConfigMaps which we call "Manifests", which are outcome of the creation of
+a CRD, these are config maps used as internal mechanisms to expose semantic data about the deployed components, more of
```

Comment:
perhaps a minor point, just for consistency:
CRD is Custom Resource _Definition_, that is - the metadata, the schema to which CR (Custom Resource) must adhere
therefore, what is being created by controller is not a CRD (CRD has already been created in the cluster) but just a CR or custom resource